src/lib/swapper/swappers/ThorchainSwapper/utils/getL1ToLongtailQuote.ts

```diff
@@ -0,0 +1,175 @@
+        code: TradeQuoteError.UnsupportedChain,
+        details: { buyAssetChainId: buyAsset.chainId },
+      }),
+    )
+  }
+
+  const chainAdapterManager = getChainAdapterManager()
+  const sellChainId = sellAsset.chainId
+  const buyChainId = buyAsset.chainId
+
+  const sellAssetFeeAssetId = chainAdapterManager.get(sellChainId)?.getFeeAssetId()
+  const sellAssetFeeAsset = sellAssetFeeAssetId ? assetsById[sellAssetFeeAssetId] : undefined
+
+  const buyAssetFeeAssetId = chainAdapterManager.get(buyChainId)?.getFeeAssetId()
+  const buyAssetFeeAsset = buyAssetFeeAssetId ? assetsById[buyAssetFeeAssetId] : undefined
+
+  if (!buyAssetFeeAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${buyChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { buyAssetChainId: buyChainId },
+      }),
+    )
+  }
+
+  if (!sellAssetFeeAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${sellChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { sellAssetChainId: sellChainId },
+      }),
+    )
+  }
+
+  const l1Tol1QuoteInput: GetTradeQuoteInput = {
+    ...input,
+    buyAsset: buyAssetFeeAsset,
+    sellAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+  }
+
+  const maybeThorchainQuotes = await getL1quote(
+    l1Tol1QuoteInput,
+    streamingInterval,
+    TradeType.L1ToLongTail,
+  )
+
+  if (maybeThorchainQuotes.isErr()) return Err(maybeThorchainQuotes.unwrapErr())
+
+  const thorchainQuotes = maybeThorchainQuotes.unwrap()
+
+  let bestAggregator: AggregatorContract
+  let quotedAmountOut: bigint
+
+  const promises = await Promise.allSettled(
+    thorchainQuotes.map(async quote => {
+      const onlyStep = quote.steps[0]
+
+      const maybeBestAggregator = await getBestAggregator(
+        buyAssetFeeAsset,
+        getWrappedToken(buyAssetFeeAsset),
+        getTokenFromAsset(buyAsset),
+        onlyStep.buyAmountAfterFeesCryptoBaseUnit,
+      )
+
+      if (maybeBestAggregator.isErr()) return Err(maybeBestAggregator.unwrapErr())
+
+      const unwrappedResult = maybeBestAggregator.unwrap()
+
+      bestAggregator = unwrappedResult.bestAggregator
+      quotedAmountOut = unwrappedResult.quotedAmountOut
+
+      const updatedMemo = addAggregatorAndDestinationToMemo({
+        aggregator: bestAggregator,
+        finalAssetAddress: fromAssetId(buyAsset.assetId).assetReference as Address,
+        minAmountOut: quotedAmountOut.toString(),
+        slippageBps: convertDecimalPercentageToBasisPoints(
+          slippageTolerancePercentageDecimal ??
+            getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Thorchain),
+        ).toString(),
+        quotedMemo: quote.memo,
+        finalAssetPrecision: buyAsset.precision,
+      })
+
+      const { data, router, vault } = await getEvmThorTxInfo({
```

Comment:
@NeOMakinG this will rug non-EVM chains - this worked for `swapIn` because inbound Txs were always EVM Txs but won't work for `swapOut` as we can have a sell Tx from any THOR-supported L1.

We will need to switch on `chainNamespace` similar to here:

https://github.com/shapeshift/web/blob/271e67b8308b04400d979f229aa2f559952d23b5/src/lib/swapper/swappers/ThorchainSwapper/utils/getL1quote.ts#L203

Reply:
Yes sorry I fucked up, I did move it one level ahead so `getL1ToLongtailQuote` becomes "chain-agnostic" and the TX info is getting computed one level upper

It's even more optimized as we don't call it multiple time now!